lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -3564,23 +3564,33 @@ namespace Js
         JavascriptMethod directEntryPoint = this->GetDefaultEntryPointInfo()->jsMethod;
         JavascriptMethod originalEntryPoint = this->GetOriginalEntryPoint_Unchecked();
 
+        FunctionBody* body = this->GetFunctionBody();
```

Comment:
This code used to check `IsFunctionBody` before calling `GetFunctionBody`. Could this possibly be called on a base `FunctionProxy`? Would we want such a call to not crash?

Reply:
It's true that it used to do `IsFunctionBody` but it also used to just call `GetFunctionBody` in the or statement
`(IsFunctionBody() && GetFunctionBody()) || GetFunctionBody()` is essentially what we had. So the `IsFunctionBody()` was pointless.

However, the only difference here is if any of the first 3 conditions were true, then we would not `GetFunctionBody()`
```cpp
IsIntermediateCodeGenThunk(directEntryPoint) || originalEntryPoint == directEntryPoint
 #if ENABLE_PROFILE_INFO
-            || (directEntryPoint == DynamicProfileInfo::EnsureDynamicProfileInfoThunk
```
However, I believe that even in those cases, `IsFunctionBody()` should be true